### PR TITLE
Never run installing user's code as root 

### PR DESCRIPTION
**Pull request by barry-scott**
Replace all cases of ${USHELL} -l with /usr/bin/su commands.
First use is commented explaining each steps purpose.

Use printenv VAR | tail -n 1 to make sure that only the VAR's value is passed on.
It is not uncommon for profiles to print information to stdout.

Refactor the MANPATH logic to use su and the tr|grep idium to look at for a path.

Delete all the logic for DISPLAY as its not required in any supported macOS
version that MacPorts supports.

Replace hardtab with spaces

### CODE REVIEW

**Comment by jmroot** [comment]
It's still possible to build an installer for Tiger (and Ryan may yet do so.)

**Comment by barry-scott** [comment]
Oh. I took that MacPorts web site making avalable only 10.9 and later kits to mean the 10.5 was not needed.

Do you need a change?

**Comment by mojca** [comment]
I cannot comment the contents, but you should avoid merging and use rebase instead.

**Comment by jmroot** [comment]
Yes, please restore the DISPLAY handling. There are packages available for OS versions back to 10.5 (I didn't build one for 10.4 because it stopped working in VirtualBox.)

**Comment by barry-scott** [comment]
@mojca I take it you mean that I should not use:
$ git fetch
$ git merge upstream/master

What do you recommend?

**Comment by jmroot** [comment]
https://trac.macports.org/wiki/WorkingWithGit#updating

tl;dr: We normally do `git pull --rebase`.

**Comment by mojca** [comment]
And potentially
```
[pull]
  rebase = true
```
in `.git/config`.

**Comment by barry-scott** [comment]
DISPLAY handling is back but with the su idiom.

**Comment by mojca** [comment]
... and then you do `git rebase -i HEAD~6` to reorder and/or squash commits, followed by a force push to branch once you are happy (this can in principle also be done by reviewer).

**Comment by barry-scott** [comment]
@mojca Are you suggesting I do that now to affect this pull request?

**Comment by neverpanic** [comment]
@barry-scott Yes, we usually bring the PR to the exact state of commits we want to apply on master before merging.

**Comment by barry-scott** [comment]
I'm not winning against git. I will close this PR and setup my git clone following your recommendations. Then generate a new PR.

**Comment by neverpanic** [comment]
@barry-scott I can also rebase the patches for you, if that helps?

**Comment by barry-scott** [comment]
@neverpanic Thanks for the offer. I did the fork again and setup up to use git merge --rebase so I should have PRs in a cleaner state from now on.
